### PR TITLE
Add some documentation to packages

### LINKED ISSUE #21637: Better API documentation for packages
### Feature Description

Currently, the package registry APIs exists under several endpoints: `/api/v1/packages`, `/api/packages`, and `/v2`. It's not very clear how they are used and what's the difference between them.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by zeripath**
In #21637 it was mentioned that the purpose of the API routes for the packages is unclear. This PR adds some documentation.

Fix #21637

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: routers/api/v1/api.go [hunk 1]
```diff
@@ -1073,6 +1073,7 @@ func Routes(ctx gocontext.Context) *web.Route {
 			}, repoAssignment())
 		})
 
+		// NOTE: these are Gitea package management API - see packages.CommonRoutes and packages.DockerContainerRoutes for endpoints to see implementations of package manager APIs
```
Comment (zeripath) [inline routers/api/v1/api.go]:
```suggestion
		// NOTE: these are Gitea package management API - see packages.CommonRoutes and packages.DockerContainerRoutes for endpoints that implement package manager APIs
```

**Comment by KN4CK3R** [comment]
While it's true that the `/v2/` routes are used for Docker the namechange is not correct. This endpoint is used for OCI compatible containers not only Docker. For example Helm uses this endpoint too and Gitea already supports this. So the old name `ContainerRoutes` may be a better match.

**Comment by zeripath** [comment]
Feel free to change it back - I guess this really shows the documentation is needed!

File: routers/api/packages/api.go [hunk 2]
```diff
@@ -40,7 +40,9 @@ func reqPackageAccess(accessMode perm.AccessMode) func(ctx *context.Context) {
 	}
 }
 
-func Routes(ctx gocontext.Context) *web.Route {
+// CommonRoutes provide endpoints for most package managers (excepting docker HUB - see below)
```
Comment (delvh) [inline routers/api/packages/api.go]:
```suggestion
// CommonRoutes provide endpoints for most package managers (except docker HUB - see below)
```

File: routers/api/packages/api.go [hunk 3]
```diff
@@ -301,7 +303,9 @@ func Routes(ctx gocontext.Context) *web.Route {
 	return r
 }
 
-func ContainerRoutes(ctx gocontext.Context) *web.Route {
+// DockerContainerRoutes provides endpoints that match the Docker HUB API in order for facilitate downloading packages a docker packages
```
Comment (delvh) [inline routers/api/packages/api.go]:
```suggestion
// DockerContainerRoutes provides endpoints that match the Docker HUB API in order to facilitate downloading docker containers
```

**Review by delvh — APPROVED** [review]

File: routers/init.go [hunk 4]
```diff
@@ -185,9 +185,15 @@ func NormalRoutes(ctx context.Context) *web.Route {
...
 	if setting.Packages.Enabled {
-		r.Mount("/api/packages", packages_router.Routes(ctx))
-		r.Mount("/v2", packages_router.ContainerRoutes(ctx))
+		// Add endpoints to match common package manager APIs
+
+		// This implements package support for most package managers
+		r.Mount("/api/packages", packages_router.CommonRoutes(ctx))
+
+		// This implements Docker HUB API (Note this is not preceded by /api but is instead /v2)
+		r.Mount("/v2", packages_router.DockerContainerRoutes(ctx))
```
Comment (lunny) [inline routers/init.go]: It's OCI compatible implementation not only Docker, so I think keeping the original name is better.

Comment (delvh) [reply routers/init.go]: OCIContainerRoutes?

Comment (lunny) [reply routers/init.go]: It maybe a better name but since this PR's emphasis is to change the documentation, I don't think this should block the PR.

Comment (KN4CK3R) [inline routers/init.go]:
```suggestion
		// This implements the OCI API (Note this is not preceded by /api but is instead /v2)
		r.Mount("/v2", packages_router.ContainerRoutes(ctx))
```

File: routers/api/packages/api.go [hunk 5]
```diff
@@ -301,7 +303,9 @@ func Routes(ctx gocontext.Context) *web.Route {
...
+// These have to be mounted on `/v2/...` due to compatibility with the Docker HUB API
+func DockerContainerRoutes(ctx gocontext.Context) *web.Route {
```
Comment (KN4CK3R) [inline routers/api/packages/api.go]:
```suggestion
// ContainerRoutes provides endpoints that implement the OCI API to serve containers
// These have to be mounted on `/v2/...` to comply with the OCI spec:
// https://github.com/opencontainers/distribution-spec/blob/main/spec.md
func ContainerRoutes(ctx gocontext.Context) *web.Route {
```

File: routers/api/packages/api.go (hunk 2 above)
Comment (KN4CK3R) [inline routers/api/packages/api.go]:
```suggestion
// CommonRoutes provide endpoints for most package managers (except containers - see below)
```

File: routers/init.go (hunk 4 above)
Comment (KN4CK3R) [reply routers/init.go]: `OCIContainerRoutes` = `OpenContainerInitiativeContainerRoutes`
I would just keep `ContainerRoutes`. The comments tell you that these are for the OCI API.

**Review by lunny — APPROVED** [review]

**Pull request merged**